tests/buffering.rs

```diff
@@ -10,6 +10,12 @@ use vector::test_util::{
 use vector::topology::{self, config};
 use vector::{buffers::BufferConfig, runtime, sinks, sources};
 
+fn make_tcp_socket_source_config(
+    tcp_config: sources::tcp::TcpConfig,
+) -> sources::socket::SocketConfig {
+    sources::socket::SocketConfig::new(sources::socket::Mode::Tcp(tcp_config))
+}
+
```

Comment:
This could be shared with `benches/buffering.rs` by moving it into `test_util.rs` (with the same comments about simplifying the calling convention as above).

Reply:
In order to make TcpConfig private I wound up adding a static method on SourceConfig make_tcp_config that takes a SocketAddr, which made this helper function obsolete.